### PR TITLE
Allow to cache columns for query based JDBC relation

### PR DESCRIPTION
**Pull request by kokosing**
Allow to cache columns for query based JDBC relation

### CODE REVIEW

**Comment by findepi** [comment]
So far there was a strong assumption that query-based relation has columns included in TH, taking the path here
https://github.com/trinodb/trino/blob/579fa3b71069a149b1d7933a3b32ead205eaa00e/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java#L135-L136

if possible, i'd like to maintain that, this would make reasoning easier.

**Comment by kokosing** [comment]
> So far there was a strong assumption that query-based relation has columns included in TH, taking the path here

So far I was no aware of this assumption. It was rather opportunistic to me. If columns are present we can use it. Disregarding if named relation or query based.

**Comment by kokosing** [comment]
> So far there was a strong assumption that query-based relation has columns included

If it is a case it should be expressed in the code, possibly as constructor check.

File: plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java [hunk 1]
```diff
@@ -478,14 +478,14 @@ CacheStats getStatisticsCacheStats()
     private static final class ColumnsCacheKey
     {
         private final JdbcIdentity identity;
-        private final SchemaTableName table;
+        private final JdbcTableHandle tableHandle;
```
Comment (ssheikin) [inline plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java:481]: @kokosing could you please doublecheck if we really want to generate cache key based on each of the fields present in `equals`/`hashCode` of `JdbcTableHandle`:
`relationHandle`, `constraint`, `sortOrder`, `limit`, `columns`, `nextSyntheticColumnId`. 
What if any of these fields is changed?
Are these fields rather constant in relation to `CachingJdbcClient#metadataCachingTtl` ?

**Pull request closed without merging**